### PR TITLE
Contribution Docs & Data Structures

### PR DESCRIPTION
**Pull request by honzajavorek**
- Reorganised contribution docs, moved them to the main docs
- Documented and demystified various data structures used in throughout Dredd, mainly those which are deep parts of the transaction object used in Dredd hooks

Related, but still not fixed: https://github.com/apiaryio/dredd/issues/599 (there's now at least place where to put it)

This change was made as part of #421 - users need to be sure what's passed over the wire to Apiary. cc also @netmilk 

**Note:** I'm not sure why coverage dropped in this case - there are no changes in code. Before rebase with `master` the coverage was green. Probably some coveralls cache?

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
 # Contributing Guidelines
 
-If you are in hurry, just please scan the **Quick Start** section.
-
 ## Quick Start
 
-### Ideas, bugs...
+### Ideas
+
+- File an [issue][issues].
```
Comment (miiila) [inline docs/contributing.md]: `Fill`?

File: docs/contributing.md [hunk 2]
```diff
@@ -94,21 +94,17 @@ When releasing, make sure you respect the tagging:
 - To release pre-release, e.g. `42.1.0-pre.7`, use just `npm publish`.
 - To release any other version, e.g. `42.1.0`, use `npm publish && npm dist-tag add dredd@42.1.0 stable`.
 
-Hopefully this will be automated one day.
+Releasing standard versions is currently automated by [Semantic Release][]. Releasing pre-releases is not automated and needs to be done manually, ideally from a special git branch.
```
Comment (miiila) [inline docs/contributing.md]: nit: I'd go with `Releasing process for {version}` instead of `Releasing {version}`.

File: docs/contributing.md [hunk 3]
```diff
@@ -94,21 +94,17 @@ When releasing, make sure you respect the tagging:
...
 It's default options are in the `test/mocha.opts` file.
 
-If you experience a flaky test, you can use `npm run test:stress` in combination
-with `describe.only` to try to replicate the flaky behavior.
-
 ### Integration Tests of Hooks Handlers
 
 Every Pull Request spawns dependent integration builds of hook handlers. Thanks
-to this author of the PR can be sure they did not break hook handler
+to this, author of the PR can be sure they did not break hook handler
```
Comment (miiila) [inline docs/contributing.md]: Either `s/author/authors` or `s/they/he`.

File: docs/data-structures.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+- startedAt (number) - unix timestamp, [transaction][].startedAt
+- title (string) - [transaction][].id
+- request (object) - [transaction][].request
+- actual (object) - [transaction][].real
+- expected (object) - [transaction][].expected
+- status (enum) - whether the validation passed or not, defaults to empty string
+  - `pass` (string)
+  - `fail` (string)
+  - `skip` (string)
+- message (string) - concatenated all messages of all [Gavel Errors](#gavel-error) in `results` or Dredd's custom message (e.g. "failed in before hook")
```
Comment (miiila) [inline docs/data-structures.md]: `Concatenation of all messages from all...`

File: docs/data-structures.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+- general (object) - contains Dredd's custom messages (e.g. "test was skipped"), formatted the same way like those from Gavel
+  - results (array[[Gavel Validator Output][]])
+- statusCode (object)
+  - results (array[[Gavel Validator Output][]])
+- headers (object)
+  - results (array[[Gavel Validator Output][]])
+- body (object)
+  - results (array[[Gavel Validator Output][]])
+- version (object, optional) - this is sometimes not present (correctly filtered out) and sometimes present
```
Comment (miiila) [inline docs/data-structures.md]: `and sometimes present` => `but sometimes is`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
 
-### :rocket: Why this change?
+#### :rocket: Why this change?
 
-### :memo: Related Issues and Pull Requests
+#### :memo: Related Issues and Pull Requests
 
-### :white_check_mark: What I did not forget?
+#### :white_check_mark: What I did not forget?
```
Comment (miiila) [inline .github/PULL_REQUEST_TEMPLATE.md]: `What I did not forget?` => `What did I not forget?`

File: docs/contributing.md (hunk 3 above)
Comment (honzajavorek) [inline docs/contributing.md]: they is a gender-neutral way how to refer to a person in modern English - https://en.wikipedia.org/wiki/Singular_they

Comment (netmilk) [inline docs/contributing.md]: Lol

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (netmilk) [inline .github/PULL_REQUEST_TEMPLATE.md]: `What didn't I forget?`

Comment (netmilk) [inline .github/PULL_REQUEST_TEMPLATE.md]: @miiila @honzajavorek `What didn't I forget?` lol

**Comment by netmilk** [comment]
This all is awesome! Thanks a lot!

**Comment by netmilk** [comment]
@honzajavorek Thanks for preserving the permalink to the Transacion Object Structure. 👍🏼

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (honzajavorek) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm not going to change it for 3rd time :D subsequent pull requests welcome

**Comment by miiila** [comment]
Based on note in description, code coverage is not considered being an issue. So... LGTM!

**Comment by honzajavorek** [comment]
Thanks! 💕

**Pull request merged**